libnavigation-ui/src/main/java/com/mapbox/navigation/ui/speed/SpeedLimitView.java

```diff
@@ -0,0 +1,213 @@
+package com.mapbox.navigation.ui.speed;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Color;
+import android.graphics.drawable.GradientDrawable;
+import android.graphics.drawable.LayerDrawable;
+import android.text.SpannableStringBuilder;
+import android.text.style.RelativeSizeSpan;
+import android.text.style.StyleSpan;
+import android.util.AttributeSet;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.lifecycle.LifecycleObserver;
+
+import com.mapbox.navigation.base.speed.model.SpeedLimit;
+import com.mapbox.navigation.base.speed.model.SpeedLimitSign;
+import com.mapbox.navigation.base.speed.model.SpeedLimitUnit;
+import com.mapbox.navigation.ui.R;
+
+import static android.graphics.Typeface.BOLD;
+import static android.text.Spanned.SPAN_EXCLUSIVE_EXCLUSIVE;
+
+public class SpeedLimitView extends ConstraintLayout implements LifecycleObserver {
+
+  private static final double KILO_MILES_FACTOR = 0.621371;
+  private static final float RADIUS = 10f;
+  private static final int VIENNA_MUTCD_BORDER_INSET = 4;
+  private static final int MUTCD_INNER_BACKGROUND_INSET = 7;
+  private static final int VIENNA_INNER_BACKGROUND_INSET = 16;
+  private static final int VIENNA_MUTCD_OUTER_BACKGROUND_INSET = 0;
+  private static final int MUTCD_BORDER_COLOR = Color.BLACK;
+  private static final int VIENNA_BORDER_COLOR = Color.RED;
```

Comment:
Can you post screenshots of what both designs look like in a night/tunnel style? I’m worried the colors will be overpowering in those dark surroundings. For comparison, the default iOS colors for [day](https://github.com/mapbox/mapbox-navigation-ios/blob/ae18aaa66a1b752dbbe62d4402a857b7a3a96f60/MapboxNavigation/DayStyle.swift#L182-L184) and [night](https://github.com/mapbox/mapbox-navigation-ios/blob/ae18aaa66a1b752dbbe62d4402a857b7a3a96f60/MapboxNavigation/DayStyle.swift#L284) are not quite red or white.